components/resources/library/src/skikoMain/kotlin/org/jetbrains/compose/resources/ImageResources.skiko.kt

```diff
@@ -4,12 +4,34 @@ import androidx.compose.ui.graphics.ImageBitmap
 import androidx.compose.ui.graphics.painter.Painter
 import androidx.compose.ui.graphics.toComposeImageBitmap
 import androidx.compose.ui.unit.Density
-import org.jetbrains.skia.Data
-import org.jetbrains.skia.Image
+import org.jetbrains.skia.*
 import org.jetbrains.skia.svg.SVGDOM
 
-internal actual fun ByteArray.toImageBitmap(): ImageBitmap =
-    Image.makeFromEncoded(this).toComposeImageBitmap()
+internal actual fun ByteArray.toImageBitmap(resourceDensity: Int, targetDensity: Int): ImageBitmap {
+    val image = Image.makeFromEncoded(this)
+
+    val targetImage: Image
+    //https://youtrack.jetbrains.com/issue/CMP-5657
+    //android only downscales drawables. If there is only low dpi resource then use it as is (not upscale)
+    //we need a consistent behavior on all platforms
+    if (resourceDensity > targetDensity) {
+        val scale = targetDensity.toFloat() / resourceDensity.toFloat()
+        val targetH = image.height * scale
+        val targetW = image.width * scale
+        val srcRect = Rect.Companion.makeWH(image.width.toFloat(), image.height.toFloat())
+        val dstRect = Rect.Companion.makeWH(targetW, targetH)
+
+        targetImage = Surface.makeRasterN32Premul(targetW.toInt(), targetH.toInt()).run {
+            val paint = Paint().apply { isAntiAlias = true }
+            canvas.drawImageRect(image, srcRect, dstRect, paint)
```

Comment:
For resizing images we also need to set SamplingMode
![image](https://github.com/user-attachments/assets/5a10d752-d4c4-43ae-b8a0-a821ccac2caa)

What will be the best for downscaling? In my experience, `nearest` produces acceptable output only with integers (2x, 3x, etc).

![image](https://github.com/user-attachments/assets/a1061466-af66-4070-9d3e-9b1ebd115c7a)

Reply:
I checked it visually and the default mode seems very close to the android downscaling result

Reply:
<img width="145" alt="image" src="https://github.com/user-attachments/assets/ef03676f-64c1-4b24-9920-0270a9e62265">

Reply:
done

Reply:
I checked it on desktop (`resourceDensity = XXXHDPI, screenDensity = HDPI`). It doesn't looks good:

[downscaleResults.zip](https://github.com/user-attachments/files/16266314/downscaleResults.zip)

I would prefer we change it to `LINEAR`, others are too slow.

```
canvas.drawImageRect(image, srcRect, dstRect, SamplingMode.LINEAR, paint, true)
```